### DISCUSSION TITLE
What is the best way to handle OTLP data and save it in DB?

### DISCUSSION THREAD

**Discussion by enono78** [open-telemetry/community#1284] [Community Discussions]
Our Collector outputs the data through an `OTLPSpanExporter`. 
We would like to save this data in a JSON format in some NoSQL DB.

What could be the best way to achieve this?

**Comment by austinlparker** [comment]
I'm not aware of a NoSQL DB that supports OTLP ingest directly, but I would imagine that it wouldn't be too hard to have them accept OTLP in JSON format through an HTTP endpoint? The exact specifics of doing this would vary depending on the database.

**Reply by enono78** [reply]
Yes, this is probably the most simple way to proceed.
Having an HTTP service that would handle the OLTP message and save it in the DB.
But I was wondering if there is already some such service, I mean it is pretty similar as a receiver no?

**Reply by austinlparker** [reply]
Like I said, not really sure. You can look at the existing exporters in collector-contrib (https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter) to see if there's something there. I see exporters for DB's like ClickHouse and Parquet. This is really more of a question for whatever database you would like to use, though.
